pkg/rke2/psp.go

```diff
@@ -324,16 +325,46 @@ func setPSPs(clx *cli.Context) func(context.Context, daemonsConfig.Control) erro
 				}
 			}
 
-			if _, err := cs.CoreV1().Namespaces().Update(ctx, ns, metav1.UpdateOptions{}); err != nil {
-				logrus.Fatalf("psp: update namespace annotation: %s", err.Error())
+			if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+				if _, err := cs.CoreV1().Namespaces().Update(ctx, ns, metav1.UpdateOptions{}); err != nil {
+					if apierrors.IsConflict(err) {
+						if err := updateNamespaceRef(ctx, cs, ns); err != nil {
+							return err
+						}
+					}
+					return err
+				}
+				return nil
+			}); err != nil {
+				logrus.Fatalf("psp: update namespace: %s - %s", ns.Name, err.Error())
 			}
-
 			logrus.Info("Applying PSP's complete")
 		}()
 		return nil
 	}
 }
 
+// updateNamespaceRef receives a value of type v1.Namespace pointer
+// and updates that value to point to a newly retrieve value in
+// the event a conflict error is returned.
+func updateNamespaceRef(ctx context.Context, cs *kubernetes.Clientset, ns *v1.Namespace) error {
+	logrus.Info("updating namespace: " + ns.Name)
+	newNS, err := cs.CoreV1().Namespaces().Get(ctx, metav1.NamespaceSystem, metav1.GetOptions{})
```

Comment:
Unless the intent is to update kube-system regardless of what namespace is passed in...
```suggestion
	newNS, err := cs.CoreV1().Namespaces().Get(ctx, ns.Name, metav1.GetOptions{})
```

Reply:
This only operates in the kube-system namespace.

Reply:
OK. Seems a little weird to pass in one NS and operate on another though. Maybe document this behavior in the doc comment?